app/src/main/java/com/ataulm/support/ColorAttributeResolvers.kt

```diff
@@ -0,0 +1,36 @@
+package com.ataulm.support
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.content.res.Resources
+import android.util.TypedValue
+import androidx.annotation.AttrRes
+import androidx.annotation.ColorInt
+import androidx.appcompat.content.res.AppCompatResources
+
+/**
+ * If it's possible to use a [ColorStateList] instead of a color int,
+ * then use [Context.resolveColorStateListAttribute] instead.
+ */
+@ColorInt
+fun Context.resolveColorAttribute(@AttrRes attr: Int): Int {
+    return resolveColorStateListAttribute(attr).defaultColor
+}
+
+fun Context.resolveColorStateListAttribute(@AttrRes attr: Int): ColorStateList {
```

Comment:
Wrote these while at Monzo but slight update - these will crash if the attribute is not found!